sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/AzureMonitorLogExporter.cs

```diff
@@ -19,18 +19,20 @@ public sealed class AzureMonitorLogExporter : BaseExporter<LogRecord>
         private readonly ITransmitter _transmitter;
         private readonly string _instrumentationKey;
         private AzureMonitorResource? _resource;
+        private readonly float _sampleRate; // This value is recorded on TelemetryItem.SampleRate.
         private bool _disposed;
 
         /// <summary>
         /// Initializes a new instance of Azure Monitor Log Exporter.
         /// </summary>
         /// <param name="options">Configuration options for the exporter.</param>
-        public AzureMonitorLogExporter(AzureMonitorExporterOptions options) : this(TransmitterFactory.Instance.Get(options))
+        public AzureMonitorLogExporter(AzureMonitorExporterOptions options) : this(options, TransmitterFactory.Instance.Get(options))
         {
         }
 
-        internal AzureMonitorLogExporter(ITransmitter transmitter)
+        internal AzureMonitorLogExporter(AzureMonitorExporterOptions options, ITransmitter transmitter)
         {
+            _sampleRate = (float)Math.Round(options.SamplingRatio * 100);
```

Comment:
can we check if distro and set samplerate? In the other part of the code we could just check null on samplerate.

Reply:
`isDistro` may not be initialized during ctor execution. It is set after the first export when resource is accessed.